solutions/usaco-736.mdx

```diff
@@ -86,120 +147,182 @@ public class BovineGeonomics {
 
 ```py
 with open('cownomics.in', 'r') as read:
-	data = [i.strip() for i in read.readlines()]
-
-n, m = map(int, data[0].split())
+	n, m = map(int, read.readline().split())
 
-spotty_cows = [data[i + 1] for i in range(n)]
-plain_cows = [data[i + n + 1] for i in range(n)]
+	# Each genome and the individual characters in each genome
+	spotted_cows = [read.readline() for _ in range(n)]
+	plain_cows = [read.readline() for _ in range(n)]
 
-total = 0
-"""
-For every position, check if it can be a potential mutation location
-and check if the spotty and plain cows have any
-repeated characters at this position
-"""
+poss_positions = 0
+# Iterate through every character
 for i in range(m):
-	used = False
+	dupe = False
+	# Iterate through every genome
 	for j in range(n):
+		# Compare to every other genome
 		for k in range(n):
-			if spotty_cows[j][i] == plain_cows[k][i]:
-				used = True
+			"""
+			If there are any duplicates,
+			then this isn't a possible position,
+			so we can stop counting and make sure we don't count it
+			"""
+			if spotted_cows[j][i] == plain_cows[k][i]:
+				dupe = True
 				break
-	if used == False:
-		total += 1
-
-print(total, file=open('cownomics.out', 'w'))
+	
+	"""
+	If we haven't found any equal characters,
+	then there are no duplicates and this is a possible sequence
+	"""
+	if not dupe:
+		poss_positions += 1
+
+print(poss_positions, file=open('cownomics.out', 'w'))
 ```
 
 </PySection>
 </LanguageSection>
 
 ## Solution 2 - Hashset
 
-<LanguageSection>
-<JavaSection>
-
 ## Implementation
 
 **Time Complexity:** $\mathcal{O}(NM)$
 
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <set>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+int main() {
+	std::ifstream read("cownomics.in");
+
+	int n;
+	int m;
+	read >> n >> m;
+	
+	vector<vector<char>> spotted_cows(n, vector<char>(m));
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < m; j++) {
+			read >> spotted_cows[i][j];
+		}
+	}
+
+	vector<vector<char>> plain_cows(n, vector<char>(m));
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < m; j++) {
+			read >> plain_cows[i][j];
+		}
+	}
+
+	int poss_positions = 0;
+	for (int i = 0; i < m; i++) {
+		std::set<char> seen;
+		for (int j = 0; j < n; j++) {
+			seen.insert(plain_cows[j][i]);
+		}
+
+		bool dupe = false;
+		for (int j = 0; j < n; j++) {
+			if (seen.count(spotted_cows[j][i])) {
```

Comment:
```suggestion
			if (seen.count(spotted_cows[j][i]) > 0) {
```

(actually we can simplify this part

```
		bool dupe = false;
		for (int j = 0; j < n && !dupe; j++) {
			dupe = seen.count(spotted_cows[j][i]) > 0;
		}
```

Reply:
i don't think that'll work bc if there's a false at the end it won't register